astropy/io/misc/asdf/tags/coords/representation.py

```diff
@@ -0,0 +1,64 @@
+from asdf.yamlutil import custom_tree_to_tagged_tree
+
+from astropy.coordinates import Longitude
+from astropy.coordinates.representation import (UnitSphericalRepresentation,
+                                                SphericalRepresentation)
+from astropy.tests.helper import assert_quantity_allclose
+
+from ...types import AstropyType
+
+
+class UnitSphericalRepresentationType(AstropyType):
+    name = "coords/representation/unitspherical"
+    types = ['astropy.coordinates.representation.UnitSphericalRepresentation']
+    requires = ['astropy']
+    version = "1.0.0"
+
+    @classmethod
+    def to_tree(cls, representation, ctx):
+        if representation.differentials:
+            raise NotImplementedError("Can not save out Representations with differentials.")
+
+        node = {}
+        node['lat'] = custom_tree_to_tagged_tree(representation.lat, ctx)
+        # Force a convert to Longitude here, it seems that asdf has some issues
+        # with subclasses thereof in `custom_tree_to_tagged_tree`
+        node['lon'] = custom_tree_to_tagged_tree(Longitude(representation.lon), ctx)
```

Comment:
@drdavella It seems that there is weridness with passing subclasses of classes that have schemas and tags. It's the same here and with the subclasses of representation in SunPy.